modules/gateways/razorpay/rzpordermapping.php

```diff
@@ -0,0 +1,73 @@
+<?php
+use Illuminate\Database\Capsule\Manager as Capsule;
+
+class RZPOrderMapping
+{
+    private $name;
+
+    function __construct($name)
+    {
+        $this->name = $name;
+    }
+
+    function createTable()
+    {
+        if (!Capsule::schema()->hasTable('tblrzpordermapping'))
+        {
+            Capsule::schema()->create('tblrzpordermapping', function($table) {
+                $table->increments('id');
+                $table->string('merchant_order_id', 30);
+                $table->string('razorpay_order_id', 50);
+            });
+        }
+    }
+
+    function insertOrder($merchant_order_id, $razorpay_order_id)
+    {
+        if (($this->validateMerchantOrderID($merchant_order_id) === false) or
+            ($this->validateRazorpayOrderID($razorpay_order_id) === false))
+        {
+            $error = array(
+                "merchant_order_id" => $merchant_order_id,
+                "razorpay_order_id" => $razorpay_order_id
+            );
+            logTransaction($this->name, $error, "Validation Failure");
+            return;
+        }
+        $insert_array = [
+            "merchant_order_id" => $merchant_order_id,
+            "razorpay_order_id" => $razorpay_order_id
+        ];
+        Capsule::table('tblrzpordermapping')->insert($insert_array);
+    }
+
+    function getRazorpayOrderID($merchant_order_id)
+    {
+        if (($this->validateMerchantOrderID($merchant_order_id)) === false)
+        {
+            $error = array("merchant_order_id"=>$merchant_order_id);
+            logTransaction($this->name, $error, "Validation Failure");
+            return;
+        }
+        $result = Capsule::table('tblrzpordermapping')
+            ->select("razorpay_order_id")
+            ->where("merchant_order_id", "=", $merchant_order_id)
+            ->orderBy('id', 'desc')
+            ->first();
+
+        return $result->razorpay_order_id;
+    }
+
+    function validateMerchantOrderID($merchant_order_id)
+    {
+        $pattern = '(^[0-9]+$)';
```

Comment:
@rohitcbr , are we sure that merchnat order Id is always in this format? please verify it once.

Reply:
@ramth05 we have checked it in local, merchant_order_id / invoice_id is of integer form

Reply:
it's not about local, my only question is there any way it can be overwritten from WHMCS settings. If not then we are good.

Reply:
@rohitcbr , please verify in case of overwritten your validations should not block it.